llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -7356,6 +7356,38 @@ void CombinerHelper::applyBuildFnMO(const MachineOperand &MO,
   Root->eraseFromParent();
 }
 
+void CombinerHelper::applyExpandFPowI(MachineInstr &MI, BuildFnTy &MatchInfo) {
+  auto [Dst, Base, Exp] = MI.getFirst3Regs();
+  LLT Ty = MRI.getType(Dst);
+
+  MachineRegisterInfo &MRI = *Builder.getMRI();
+  std::optional<int64_t> ConstantExpValue = getIConstantVRegSExtVal(Exp, MRI);
+  assert(ConstantExpValue && "matched non-const FPOWI?");
+
+  int64_t OriginalExprVal = *ConstantExpValue;
+  int64_t ExpVal = OriginalExprVal;
+
```

Comment:
```
if (!MRI.hasOneNonDBGUse(Base))
  return false;
```

Reply:
Why is this needed? Why can't we optimize if there are multiple uses of Base?

Reply:
@tschuett sorry for the ping, but can you please help me understand why this change is needed, so I can push the PR to completion?

Reply:
Ignore that comment.